tests/models/test_signature.py

```diff
@@ -338,3 +339,42 @@ def test_infer_signature_with_dataclass():
     output_schema = convert_dataclass_to_schema(rag_signatures.ChatCompletionResponse())
     assert inferred_signature.inputs == input_schema
     assert inferred_signature.outputs == output_schema
+
+
+@dataclass
+class CustomInput:
+    id: int = 0
+
+
+@dataclass
+class CustomOutput:
+    id: int = 0
+
+
+@dataclass
+class FlexibleChatCompletionRequest(rag_signatures.ChatCompletionRequest):
+    custom_input: Optional[CustomInput] = None
+
+
+@dataclass
+class FlexibleChatCompletionResponse(rag_signatures.ChatCompletionResponse):
+    custom_output: Optional[CustomOutput] = None
+
+
+def test_infer_signature_with_optional_dataclass():
+    inferred_signature = infer_signature(
+        FlexibleChatCompletionRequest(), FlexibleChatCompletionResponse()
+    )
+    custom_input_schema = next(
+        schema for schema in inferred_signature.inputs.to_dict() if schema["name"] == "custom_input"
+    )
+    assert custom_input_schema["required"] is False
+
+
+def test_infer_signature_with_child_dataclass():
```

Comment:
I think we can combine this test with the above one into one. And another test case for validating `FlexibleChatCompletionRequest(custom_input=CustomInput())` can capture `id` as well

Reply:
changed!